lib/actions-oclif/device/move.ts

```diff
@@ -82,47 +85,67 @@ export default class DeviceMoveCmd extends Command {
 		options.application = options.application || options.app;
 		delete options.app;
 
-		const device: ExtendedDevice = await balena.models.device.get(
-			params.uuid,
-			expandForAppName,
+		const devices: ExtendedDevice[] = await Promise.all(
+			params.uuid
+				.split(',')
+				.map((uuid) => balena.models.device.get(uuid, expandForAppName)),
 		);
 
-		const belongsToApplication = device.belongs_to__application as Application[];
-		device.application_name = belongsToApplication?.[0]
-			? belongsToApplication[0].app_name
-			: 'N/a';
+		for (const device of devices) {
+			const belongsToApplication = device.belongs_to__application as Application[];
+			device.application_name = belongsToApplication?.[0]
+				? belongsToApplication[0].app_name
+				: 'N/a';
+		}
 
 		// Get destination application
 		let application;
 		if (options.application) {
 			application = options.application;
 		} else {
-			const [deviceDeviceType, deviceTypes] = await Promise.all([
-				balena.models.device.getManifestBySlug(device.device_type),
+			const [deviceDeviceTypes, deviceTypes] = await Promise.all([
+				Promise.all(
+					devices.map((device) =>
+						balena.models.device.getManifestBySlug(device.device_type),
+					),
+				),
 				balena.models.config.getDeviceTypes(),
 			]);
 
-			const compatibleDeviceTypes = deviceTypes.filter(
-				(dt) =>
-					balena.models.os.isArchitectureCompatibleWith(
-						deviceDeviceType.arch,
-						dt.arch,
-					) &&
-					!!dt.isDependent === !!deviceDeviceType.isDependent &&
-					dt.state !== 'DISCONTINUED',
+			const compatibleDeviceTypes = deviceTypes.filter((dt) =>
+				deviceDeviceTypes.every(
```

Comment:
I am contributing a commit: please review it, test the command and feel free to make any further changes (and let me know when/if you're happy. I was doing some testing myself and, before my commit, I realised that some corner cases existed if the device list mixed devices of incompatible architectures. Example, before my commit:
```
$ ./bin/balena device move b1878c6,dcbff32
? Select an application
❯ test-19282 (qemux86-64)
  test-nuc (intel-nuc)

$ ./bin/balena device move dcbff32,b1878c6
? Select an application
  tests (raspberrypi3)
  test-skx2 (skx2)
  test-tx2 (jetson-tx2)
❯ bbb-test (beaglebone-green-wifi)
  essentials (raspberrypi3)
  iwantlegacy (raspberrypi3)
  raspberrypi2 (raspberry-pi2)
```
Where b1878c6 is an amd64 device, dcbff32 is an ARM device. Note how the list of applications changes depending on the order of the devices in the comma-separated list in the command line. By the way, I realise that it was my own suggestion to use `devices[0]`, then again, the original code before my suggestion was also subject to this issue. :-)
After my commit:
```
$ ./bin/balena device move dcbff32,b1878c6
No suitable applications found for selection
Do all devices have a compatible architecture?

$ ./bin/balena device move dcbff32,b1878c6 -a test-rpi
dcbff32 was moved to test-rpi
BalenaInvalidDeviceType: Invalid device type: Incompatible application: test-rpi
```
The last case above raises `BalenaInvalidDeviceType`, which I had added to `errors.ts` as an "expected error". I think it is acceptable.